src/test/java/com/puppycrawl/tools/checkstyle/BaseAstTestSupport.java

```diff
@@ -0,0 +1,98 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class BaseAstTestSupport {
```

Comment:
> CheckTest support class

It sorta is, but can be used for other things too. For example, `AbstractViolationReporterTest` uses it.
Also, if another test extends `BaseCheckTestSupport` and it needs this method, we now have a conflict as 1 class can't extend 2 others.
Current `BaseCheckTestSupport` also already does Java grammar verification, so it seems ok to use it for Javadoc. So I still think we should use `BaseCheckTestSupport` over a new class.